mlir/lib/Dialect/Rock/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -356,89 +238,9 @@ static ArrayAttr ldsVectorLayout(OpBuilder &b, Location loc,
   return b.getArrayAttr({ignoreTidAttr});
 }
 
-static TransformingForOp
-createGlobalLoadLoop(PatternRewriter &b, Location loc, GpuAllocOp loadBuffer,
-                     Value wrappedMatrix, ArrayAttr vectorMap,
-                     int64_t dataPerThread, int64_t vectorLen, Value bid,
-                     Value tid, layout::GridCoordinates coords,
-                     bool forceUnroll) {
-  Value tensor;
-  ArrayAttr matrixToTensor;
-  std::tie(tensor, matrixToTensor) = untransform(b, wrappedMatrix);
-
-  // Optimize the transform chain.
-  ArrayRef<int64_t> tensorSize = tensor.getType().cast<ShapedType>().getShape();
-  matrixToTensor = collapseContiguousMerges(matrixToTensor, tensorSize);
-
-  Type elementType =
-      wrappedMatrix.getType().cast<MemRefType>().getElementType();
-  Type loadType = vectorTypeOrSelf(elementType, vectorLen);
-  Value zero = b.createOrFold<arith::ConstantIndexOp>(loc, 0);
-
-  SmallVector<Value, 4> globalStart = {
-      zero, coords.g_block, coords.m_block, coords.n_block, tid, zero};
-  SmallVector<Value, 4> vectorStartOuter(6, zero);
-  auto outerLoop = b.create<TransformingForOp>(
-      loc, ArrayRef<ValueRange>{globalStart, vectorStartOuter},
-      ArrayRef<Attribute>{matrixToTensor, b.getArrayAttr({})},
-      /*bounds=*/ArrayRef<int64_t>{1, 1, 1, 1, 1, dataPerThread},
-      /*strides=*/ArrayRef<int64_t>{1, 1, 1, 1, 1, vectorLen}, forceUnroll,
```

Comment:
Also I'd like to point out that vectorLen is already forced in the current code where there isn't vec len inference done after untransform. So this PR doesn't change this aspect.

Reply:
rocmlir-gen --operation conv2d -t i8 --arch gfx908:sramecc+:xnack- --num_cu 120 --fil_layout kcyx --in_layout nchw --out_layout nkhw --batchsize 256 --in_channels 64 --in_h 56 --in_w 56 --out_channels 256 --fil_h 1 --fil_w 1 --dilation_h 1 --dilation_w 1 --conv_stride_h 1 --conv_stride_w 1 --padding_h 0 --padding_w 0 --kernel-repeats 5 --perf_config=32,128,32,32,32,1,1,1 | rocmlir-driver -c --mlir-print-ir-before=rock-blockwise-gemm-to-threadwise &> dump_br.log